code/datums/antagonists/datum_changeling.dm

```diff
@@ -0,0 +1,60 @@
+/datum/antagonist/changeling
+	name = ROLE_CHANGELING
+
+	text_on_lose = "You are no longer a changeling!"
+
+	give_special_equipment = FALSE
+
+	number_of_possible_objectives = 2
+	possible_objectives = list(/datum/objective/steal, /datum/objective/assassinate, /datum/objective/maroon)
+
+/datum/antagonist/changeling/apply_innate_effects()
+	if(!owner)
+		return
+	if(owner.current)
+		owner.current.make_changeling()
+
+	ticker.mode.update_changeling_icons_added(owner)
```

Comment:
Yea I am going to do something with huds, but I didn't get to it yet(the generalizing part with args was my first idea anyway).